src/main/java/seedu/todo/controllers/ListController.java

```diff
@@ -78,522 +80,147 @@ public void process(String input) throws ParseException {
-                    }
-                } else {
-                    return (List<Event>) null;
-                }
-            } else if (isCompleted) {
-                return db.getEventByRange((LocalDateTime) null, LocalDateTime.now());
-            } else {
-                return db.getEventByRange(LocalDateTime.now(), (LocalDateTime) null);
-            }
-        } else if (dateOn != null) { //by keyword found
-            return db.getEventByDate(dateOn);
-        } else {
-            return db.getEventByRange(dateFrom, dateTo);
-        }
-    }
-
-    /**
-     * Setting up Task view 
-     * 
-     * @param isCompleted
-     *            true if user request completed item
-     * @param listAllStatus
-     *            true if Status is not provided, isCompleted values will be ignored     
-     * @param dateOn
-     *            Date if user specify for a certain date
-     * @param dateFrom
-     *            Due date for Task or start date for Event
-     * @param dateTo
-     *            End date for Event  
-     * @param isDateProvided 
-     *            true if date is provided is any part of the input            
-     * @param isExactCommand
-     *            true if input is exactly the same as COMMAND_WORD
-     * @param listAll
-     *            true if CalendarItem Type not provided
-     * @param db
-     *            instance of TodoListDB db                              
-     */
-private List<Task> setupTaskView(boolean isCompleted, boolean listAllEventStatus, boolean listAllTaskStatus, LocalDateTime dateOn, LocalDateTime dateFrom,
-            LocalDateTime dateTo, boolean isDateProvided, boolean isExactCommand, boolean listAll, TodoListDB db) {
-        if (!listAllEventStatus) {
-            return null;
-        }
-        if (dateFrom == null && dateTo == null && dateOn == null) {
-            if (listAllTaskStatus) { // not specify
-                if (isExactCommand && isDateProvided == false) {
-                    if (listAll) {
-                        return db.getIncompleteTasksAndTaskFromTodayDate();
-                    } else {
-                        return db.getAllTasks();
-                    }
-                } else {
-                    return (List<Task>) null;
-                }
-            } else {
-                return db.getTaskByRangeWithStatus(dateFrom, dateTo, isCompleted, listAllTaskStatus);
-            }
-        } else if (dateOn != null) { //by keyword found
-            return db.getTaskByDateWithStatus(dateOn, isCompleted, listAllTaskStatus);
-        } else {
-            return db.getTaskByRangeWithStatus(dateFrom, dateTo, isCompleted, listAllTaskStatus);
-        }
-    }
-    
-    /** ================ PARSING METHODS ================== **/
-    
-    /**
-     * Parse a natural date into a LocalDateTime object.
-     * 
-     * @param natural
-     * @return LocalDateTime object
-     */
-    private LocalDateTime parseNatural(String natural) {
```

Comment:
yes, I do agree that factoring out the code will do justice to the cleanliness to the code and i do agree that this might be a good working base which could have been adapted, but as replied in my telegram msg, I asked this "did u comm with ur team when u did tht?" similar to my abv question "have ur been communicating while* you code" and my reason given as clearly stated - " If i'm not wrong, when ur do a main merge, it'll result to a big bang - reason? the file changes are too drastic." and as replied in my telegram chat to you, it is "not recommended at this stage" bcos there is 2.5k worth of line of code change and it is difficult for ur teammate to just do a merge, since you might not be the one doing the merging and fixing u might not understand the amt of hours that requires one to review this code, do put urself in their shoes what if you were in this situation now and you are running out of time? If this results to a major breakdown in your entire project, will it be worth? plus you'll have to redo alot of testing, though you might have already done some according to ur commits above but your coverage might decrease bcos now there is more path to run through, **unless of course the change is really clean and you can guarantee in every sense that other places will not break then this commit will be fine.** 

In addition you might not be even able to claim it under yours since the codes you have refactored are mainly written by your teammates according to the commits I've checked on github, it is not fair towards them if you claim the entire code if you've only done refactoring instead of writing the logics yourself.

@irvinlim @louietyj @ChaseYaoCong  please resolve this issue asap since monday is the deadline, regardless of what you guys agree on in the end I do hope everything turn out fine. do keep me in the loop and All the best.